scipy/spatial/transform/rotation.py

```diff
@@ -246,3 +247,37 @@ def from_rotvec(cls, rotvec):
             return cls(quat[0], normalized=True)
         else:
             return cls(quat, normalized=True)
+
+    def as_rotvec(self):
+        """Return the rotation vector representation of the Rotation.
+
+        This function returns a numpy.ndarray of shape (3,) or (N, 3)
+        depending on the input that was used to initialize the object.
+
+        A rotation vector is a 3 dimensional vector which is co-directional to
+        the axis of rotation and whose norm gives the angle of rotation (in
+        radians).
+        """
+        quat = self._quat.copy()
+        # w > 0 to ensure 0 <= angle <= pi
+        quat[quat[:, 3] < 0] *= -1
+
+        angle = 2 * np.arctan2(np.linalg.norm(quat[:, :3], axis=1), quat[:, 3])
+
+        small_angle = (angle <= 1e-3)
```

Comment:
Just FYI, the parenthesis are not strictly necessary here as far as I know.

Question to @larsoner --- should we ever worry about boolean mask vs indices performance? I mean we can compute indices using a single `nonzero` call and then use them.

Reply:
I don't know the answer offhand, it might change based on the use case (best to test if necessary). In this instance the code is simplified by a boolean mask which is more important in my mind, at least in first implementation.

In terms of performance in this particular instance, doing `np.nonzero` followed by use of the returned indices should be slower than using the `bool` mask directly, since both will require iterative logical evaluation of the mask (to determine in one case the indices, and in the other which values to fill in the assignment), but the `np.nonzero` will also require an additional memory allocation (for the indices themselves) and iteration (over the indices when filling the array). This is assuming NumPy under the hood does something close to optimal in terms of evaluations, at least, which it probably does.

Feel free to use `timeit` to test these things directly, though, if you want.

Reply:
And yes the parens are not strictly necessary. I find it easier as personal preference to read code as `a = (...)` rather than `a = ...` when the `...` is a boolean evaluation, because it's obvious if you see a `=` contained within the `(...)` it cannot be an assignment operator. In other words it is possible to do `a = b = c`, `a = b == c`, and `a = (b == c)`, but not `a = (b = c)`, so the parens do actually add information (and thus readability IMO).

Reply:
I also thought that the only way to find out is to use timeit. I only note that here boolean mask is used 3 times after creation, so in theory there can be a benefit of using indices. I'm quite sure that it won't speed up anything significantly (if at all). And the code is better this way.

For parentheses usage --- totally makes sense to me as well.